src/validation/validation.cpp

```diff
@@ -1743,6 +1743,11 @@ uint32_t GetBlockScriptFlags(const CBlockIndex *pindex, const Consensus::Params
         flags |= SCRIPT_VERIFY_NULLFAIL;
     }
 
+    // BU: Set enableCheckDataSigVerify flag depending on whether CDS/-V should be enabled for this block
+    // This is a nasty side-effect kind of way of doing this and should be replaced
+    // with passing the flag to or setting it in a script verification class.
+    enableCheckDataSigVerify = IsNov152018Enabled(consensusparams, pindex->pprev);
```

Comment:
Why `pindex->pprev`? `IsNov152018Enabled already check if median time past (`MTP`) is grater or equal of activation time for current tip and tip - 1 (i.e. `pprev). 
That way it checks for tip-2 and tip-1.

Reply:
That sounds right, thank you very much! I am not really firm in the matter of hard fork switch-over points and also I am confused by ABC's seemingly different logic.

As a further TODO, the boundary should be checked in a sharp sense and I had the `checkdatasig-activation.py` code passing, which it shouldn't given the right conditions, correct?

Reply:
IMHO ABC and BU behave the same even if the way IsNov152018enabled and IsMagnetictAnomalyEnabled seem different.

As I said the former check from MTP trigger on both pindex and pindex-1 whereas the latter check only pindex this is why ABC is (probably, didn't check the code) passing pprev rather then the current tip.